### PR TITLE
Add share_reapply task - ON DEMAND for data.all admins

### PR DESCRIPTION
**Pull request by dlpzx**
### Feature or Bugfix
- Feature

### Detail
Implements https://github.com/data-dot-all/dataall/issues/1150
- Add **on-demand** ECS task to reapply shares
- It goes through ALL active shares, if there are unhealthy items, it reapplies the share

As explained in the issue description, this task is a resource of data.all admins that can run it on-demand using ECS commands or directly in the AWS console. We could extend how it is used:
- By triggering it in the `share-verifier` task daily. We can add a one-liner that triggers the share-reapplier task at the end of the share-verifier task.
- By adding functionalities in the UI that allow users to trigger this task

### Testing
Deploying in AWS:
- [X] CICD pipeline runs successfully and creates new on-demand ECS task definition `share-reapplier`
- [X] We can run the ECS task `share-reapplier` from the AWS console
- [X] Checking the logs of the ECS task `share-reapplier` without any unhealthy share succeeds
- [X] Checking the logs of the ECS task `share-reapplier` with unhealthy shares succeeds - reapplies the unhealthy shares

### Relates
- #1150 
- #1064 

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: backend/dataall/modules/dataset_sharing/tasks/share_reapplier_task.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from dataall.base.db import get_engine
+
+root = logging.getLogger()
+root.setLevel(logging.INFO)
+if not root.hasHandlers():
+    root.addHandler(logging.StreamHandler(sys.stdout))
+log = logging.getLogger(__name__)
+
+
+def verify_shares(engine):
```
Comment (noah-paige) [inline backend/dataall/modules/dataset_sharing/tasks/share_reapplier_task.py:17]: change name of func and L44

**Comment by dlpzx** [comment]
Hi @noah-paige where do you think we should include the documentation about this feature? In the userguide, in the github pages or in both? I was going to go for both

**Comment by noah-paige** [comment]
> Hi @noah-paige where do you think we should include the documentation about this feature? In the userguide, in the github pages or in both? I was going to go for both

Can't hurt to do both! I was thinking definitely in the UserGuide under the shares page where we have a section discussing verify and re-apply of shares

For GH Pages if you think there is a good place to add docs as well I think it would be helpful

**Review by noah-paige — APPROVED** [review]
Tested deploy and running the reapply task on demand - all looks good!

Ideally we would have a way to default to the correct VPC, Private Subnet, and Security Group selections but looking at CDK docs I am not sure that is possible

As long as we have good documentation of how to run task ad-hoc and how to retrieve the network configuration (maybe via SSM getParam calls?) then I think we are good with this issue

**Comment by dlpzx** [comment]
> with this i
There are ways of getting the parameters programmatically, because the networking needs to be specified in the runTask API call. I'll include the docs in a separate PR

**Pull request merged**